src/main/java/ca/fxco/pistonlib/blocks/pistons/basePiston/BasicPistonArmBlock.java

```diff
@@ -120,44 +101,39 @@ public VoxelShape getShape(BlockState state, BlockGetter world, BlockPos pos, Co
 
     public boolean isAttached(BlockState state, BlockState behindState, BlockState frontState) {
         // Must be BasicPistonArmBlock or BasicPistonHeadBlock in front
-        if (!frontState.is(HEAD_BLOCK) && !frontState.is(this)) {
+        if (!frontState.is(this.family.getHead()) && !frontState.is(this)) {
             return false;
         }
 
-        PistonType type = state.getValue(TYPE);
         Direction facing = state.getValue(FACING);
+        PistonType type = state.getValue(TYPE);
 
         // Head or Arm in front of arm is valid
-        if (type != frontState.getValue(TYPE) || facing != frontState.getValue(FACING)) {
+        if (facing != frontState.getValue(FACING) || type != frontState.getValue(TYPE)) {
```

Comment:
The arm block no longer has the piston type property yet checks for it here.

Reply:
not just there xD